### ISSUE TITLE
Timed hold should not be cancelled when the CAVC window task is assigned to a user

### ISSUE THREAD

**Issue by yoomlam** [department-of-veterans-affairs/caseflow#15893]
## Description

BUG: When "Assign to person" action is used for a `CavcRemandProcessedLetterResponseWindowTask` that is assigned to the CAVC org that is on hold, the timed hold is cancelled. The timed hold should continue with the same end time.

Identified while working on PR #15885.

## Acceptance criteria
- [ ] timed hold should continue with the same end time when window task is assigned *or reassigned* to a user

## Background/context/resources
[Slack thread](https://dsva.slack.com/archives/CJL810329/p1612828258213000)

[Interaction with extension requests](https://dsva.slack.com/archives/CJL810329/p1612910935227100)
> when granting an extension, it cancels the previous hold and start a completely new one.
> If denying an extension, it cancels the previous hold (regardless of whether the hold was created by putting it on hold or a granting an extension).

Basically, it appears only 1 `TimedHoldTask` is active at a time.

## Technical notes

If we want the timed hold to not be cancelled, we can
1. prevent cancellation of the TimedHoldTask under the org task (unclear what kind of notification the assignee of the user task would get, if any, when the timed hold is complete)
2. copy the TimedHoldTask every time the task is assigned/reassigned, with the remaining hold time
3. ~prevent creation and assignment of user tasks~ Conflicts with requirement to assign tasks to user

Option 2 seems most straightforward.

**Comment by ajspotts** [comment]
[what is this chart?](https://github.com/department-of-veterans-affairs/caseflow/issues/11533)
```bash
1 | 
2 | ||||||
3 | ||||||
5 | 
8 | 
```

There may be existing Task.rb behavior around reassign saving children

#### Why 2?
* Find and use existing solution

#### Why 3?
* pessimism
* May be tricky to debug

**Comment by yoomlam** [comment]
Going with new option 4: move TimedHoldTask from the org-task to the user-task.
Why?
* The TimedHoldTask is not really cancelled, so let's not cancel it at all.
* In case the user-task is cancelled, the TimedHoldTask can be moved back to the org-task. Basically the TimedHoldTask can move between the org-task and user-task.

**Comment by yoomlam** [comment]
Reminder to test:
* "Re-assign to person"
* "End hold early" in order to "Mark task complete" (don't need to move TimedHoldTask back to org-task b/c TimedHoldTask is required to be completed in order to "Mark task complete")

Question: Do we want user to be able to cancel a user-task?
